pkg/scheduler/apis/config/testing/compatibility_test.go

```diff
@@ -1551,3 +1551,284 @@ func TestAlgorithmProviderCompatibility(t *testing.T) {
 		})
 	}
 }
+
+func TestPluginsConfigurationCompatibility(t *testing.T) {
+	// Add serialized versions of scheduler config that exercise available options to ensure compatibility between releases
+	defaultPlugins := map[string][]config.Plugin{
+		"QueueSortPlugin": {
+			{Name: "PrioritySort"},
+		},
+		"PreFilterPlugin": {
+			{Name: "NodeResourcesFit"},
+			{Name: "NodePorts"},
+			{Name: "InterPodAffinity"},
+		},
+		"FilterPlugin": {
+			{Name: "NodeUnschedulable"},
+			{Name: "NodeResourcesFit"},
+			{Name: "NodeName"},
+			{Name: "NodePorts"},
+			{Name: "NodeAffinity"},
+			{Name: "VolumeRestrictions"},
+			{Name: "TaintToleration"},
+			{Name: "EBSLimits"},
+			{Name: "GCEPDLimits"},
+			{Name: "NodeVolumeLimits"},
+			{Name: "AzureDiskLimits"},
+			{Name: "VolumeBinding"},
+			{Name: "VolumeZone"},
+			{Name: "InterPodAffinity"},
+		},
+		"PreScorePlugin": {
+			{Name: "InterPodAffinity"},
+			{Name: "DefaultPodTopologySpread"},
+			{Name: "TaintToleration"},
+		},
+		"ScorePlugin": {
+			{Name: "NodeResourcesBalancedAllocation", Weight: 1},
+			{Name: "ImageLocality", Weight: 1},
+			{Name: "InterPodAffinity", Weight: 1},
+			{Name: "NodeResourcesLeastAllocated", Weight: 1},
+			{Name: "NodeAffinity", Weight: 1},
+			{Name: "NodePreferAvoidPods", Weight: 10000},
+			{Name: "DefaultPodTopologySpread", Weight: 1},
+			{Name: "TaintToleration", Weight: 1},
+		},
+		"BindPlugin": {{Name: "DefaultBinder"}},
+	}
+
+	testcases := []struct {
+		name          string
+		pluginsConfig config.Plugins
+		wantPlugins   map[string][]config.Plugin
+	}{
+		{
+			name:        "No plugins specified",
+			wantPlugins: defaultPlugins,
+		},
+		{
+			name: "Disable all default plugins",
+			pluginsConfig: config.Plugins{
```

Comment:
How come this doesn't produce an error? We require one QueueSort plugin and one Bind plugin.

Reply:
It's interesting, see https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/v1alpha1/framework.go#L201-L204
When there is no plugins at all, QueueSort and Bind check is bypassed.

Reply:
oh.... I remember this. I think we did it to make it easy to build an empty framework on some tests. Or maybe this was before we did queue and bind mandatory. I'd say we should remove it? WDYT @Huang-Wei

Reply:
If we need to remove it, I can simply do it in this pr, and add some tests to cover this behavior.

Reply:
Please do so, but only if not too many lines change.
But keep it in a separate commit for clear separation and also in case we want to revert it.

Reply:
Sure

Reply:
It seems that removing this logic doesn't have effects on other codes and tests.

Reply:
That's such good news